### PR TITLE
Tic tac toe middleware

### PR DESCRIPTION
**Pull request by marcalmon**
## Summary

- Completed the card of middlewares.

## Trello Card

https://trello.com/c/UEetpjLd/14-redux-ii-middleware

### CODE REVIEW

File: react/tic-tac-toe/src/app/screens/Game/components/Matches/index.js [hunk 1]
```diff
@@ -1,28 +1,22 @@
...
   }
 
   render() {
-    const { loading, data } = this.state;
+    const { loading, data } = this.props;
     return (
       <Fragment>
         <h1>Match History</h1>
-        {loading ? (
+        {loading && loading ? (
```
Comment (nicolobruno) [inline react/tic-tac-toe/src/app/screens/Game/components/Matches/index.js]: I think with one loading is sufficient

File: react/tic-tac-toe/src/redux/game/actions.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+export const actions = {
+  ADD_MOVE: '@@GAME/ADD_MOVE',
+  REMOVE_MOVE: '@@GAME/REMOVE_MOVE',
+  SET_WINNER: '@@GAME/SET_WINNER'
+};
+
+const actionsCreators = {
```
Comment (nicolobruno) [inline react/tic-tac-toe/src/redux/game/actions.js]: `actionsGame`

File: react/tic-tac-toe/src/redux/matches/actions.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import MatchesService from '../../services/MatchesService';
+
+export const actions = {
+  GET_MATCHES: '@@MATCHES/GET_MATCHES',
+  GET_MATCHES_SUCCESS: '@@MATCHES/GET_MATCHES_SUCCESS',
+  GET_MATCHES_FAILURE: '@@MATCHES/GET_MATCHES_FAILURE'
+};
+
+const actionsCreators = {
```
Comment (nicolobruno) [inline react/tic-tac-toe/src/redux/matches/actions.js]: `actionsMatches`

File: react/tic-tac-toe/src/app/screens/Game/index.js [hunk 4]
```diff
@@ -1,71 +1,53 @@
 import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { func, bool, string, number, arrayOf } from 'prop-types';
+
+import actionsCreators from '../../../redux/game/actions';
```
Comment (nicolobruno) [inline react/tic-tac-toe/src/app/screens/Game/index.js]: `actionsGame`

File: react/tic-tac-toe/src/app/screens/Game/components/Matches/index.js [hunk 5]
```diff
@@ -1,24 +1,18 @@
 import React, { Component, Fragment } from 'react';
+import { arrayOf, bool, func, shape, string } from 'prop-types';
+import { connect } from 'react-redux';
 import Spinner from 'react-spinkit';
 
-import MatchesService from '../../../../../services/MatchesService';
+import actionsCreators from '../../../../../redux/matches/actions';
```
Comment (nicolobruno) [inline react/tic-tac-toe/src/app/screens/Game/components/Matches/index.js]: `actionsMatches`

File: react/tic-tac-toe/src/app/screens/Game/components/Matches/constants.js [hunk 6]
```diff
@@ -1,2 +0,0 @@
-export const PLAYER_ONE = 'player_one';
```
Comment (nicolobruno) [inline react/tic-tac-toe/src/app/screens/Game/components/Matches/constants.js]: Why delete constants ?

File: react/tic-tac-toe/src/app/screens/Game/components/Matches/index.js [hunk 7]
```diff
@@ -30,8 +22,8 @@ class Matches extends Component {
           <ol>
             {data.map(item => (
               <li key={item.id}>
-                <span className={getWinnerClass(item.winner === PLAYER_ONE)}>{item.player_one}</span> -{' '}
-                <span className={getWinnerClass(item.winner === PLAYER_TWO)}>{item.player_two}</span>
+                <span className={getWinnerClass(item.winner === 'player_one')}>{item.player_one}</span> -{' '}
+                <span className={getWinnerClass(item.winner === 'player_two')}>{item.player_two}</span>
```
Comment (mariadelacamara) [inline react/tic-tac-toe/src/app/screens/Game/components/Matches/index.js]: why did you delete constants file and put `players` as a string again?
